lib_eio/core/eio__core.mli

```diff
@@ -403,11 +436,13 @@ module Private : sig
   module Fiber_context : sig
     type t
 
+    type vars
```

Comment:
Since this is abstract and only used as an input, it can probably be removed. I guess the `make` function doesn't need to be exposed either.